app/course.service.ts

```diff
@@ -1,29 +1,40 @@
 import {Injectable} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+import {of} from 'rxjs/observable/of';
 import {Course} from './course';
 import {COURSES} from './mock';
 
 @Injectable()
 
 export class CourseService {
+    isAddCoursePage: boolean;
     constructor() {}
 
-    getList(): Course[] {
-        return COURSES;
+    getList(): Observable<Course[]> {
+        return of(COURSES);
     }
 
-    createCourse(): void {}
+    createCourse(): void {
+        this.isAddCoursePage = true;
+    }
 
     getItemById(id: number): void {}
 
     updateItem(id: number): void {}
 
-    removeItem(id: number): void {
+    isCreatingCourse(): boolean {
+        return this.isAddCoursePage;
+    }
+
+    removeItem(id: number): Observable<Course[]> {
         COURSES.find((course, index) => {
             if (course.id === id) {
                 COURSES.splice(index, 1);
 
                 return true;
             }
         });
+
+        return of(COURSES);
```

Comment:
it's bad to create a new source to which everyone should subscribe on changes. ideally previous source should be reused. consider the case when you have one service that shares data among multiple components. One component that triggered this `removeItem()` will know that you would have updates in data. Others will not know that. This would lead to data inconsistency and bugs in code. Also, those old subscriptions would live forever introducing memory leaks.

So you should notify every subscriber of your changes. For that store only one observable in the service. Let everyone to subscribe to that observable. When data changes notify all subscribers about the changes. Use Subject to be able to implicitly call `.next()` or introduce another observable with action flow like in redux (but that is more complex than working with a Subject).